src/io/output_time.cc

```diff
@@ -91,12 +102,45 @@ OutputTime::~OutputTime(void)
 
     if (this->_base_file.is_open()) this->_base_file.close();
 
-     xprintf(MsgLog, "O.K.\n");
+    if(output_mesh_ != nullptr) delete output_mesh_;
+    xprintf(MsgLog, "O.K.\n");
 }
 
 
 
+void OutputTime::make_output_mesh(Mesh* mesh, FieldSet* output_fields)
+{
+    if(is_output_mesh_valid_) return;
+    
+    // possibly destroy previous output mesh
+    if(output_mesh_) delete output_mesh_;
+    
+    output_mesh_ = new OutputMesh(mesh);
+    
+    FieldCommon* field =  output_fields->field(error_control_field_name);
+    if( field && (typeid(*field) == typeid(Field<3,FieldValue<3>::Scalar>)) ) {
+        Field<3,FieldValue<3>::Scalar>* error_control_field = static_cast<Field<3,FieldValue<3>::Scalar>*>(field);
+        DBGMSG("Output mesh will be refined according to field '%s'.\n", error_control_field_name.c_str());
+        // create refined output mesh according to the field
+        output_mesh_->create_refined_mesh(error_control_field);
+    }
+    else{
+        DBGMSG("Output mesh identical with the computational one.\n");
```

Comment:
OK. Then I would rather suggest to create a simple record for output mesh inside output stream record. When we implement the refinement, I think it would good to be able to control some setting of the refinement (max levels, interpolation error tolerance, type of refinement - it might be useful to just set uniform refinement of given level instead of error control - occurred to me during testing refinement..)